ui/src/app/components/base/modal/conversation-telemetry-modal/index.tsx

```
@@ -224,44 +373,94 @@ export function ConversationTelemetryDialog(
+
+        mergedRows.sort((a, b) => a.ts.getTime() - b.ts.getTime());
+        setRows(mergedRows);
+        setTotalItem(total);
+      } catch {
         if (!active) return;
         setRows([]);
         setTotalItem(0);
-      })
-      .finally(() => {
+      } finally {
         if (!active) return;
```

Biome (2.4.13)

[error] Unsafe usage of 'return'.

(lint/correctness/noUnsafeFinally)